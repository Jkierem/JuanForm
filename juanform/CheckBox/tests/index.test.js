"use strict";

var _react = _interopRequireDefault(require("react"));

var _sinon = _interopRequireDefault(require("sinon"));

var _enzyme = require("enzyme");

var _toolbox = require("../../Utils/toolbox");

var _ = _interopRequireDefault(require("../"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

describe("#Checkbox", function () {
  describe("Checkbox behavior", function () {
    var mockEvent = (0, _toolbox.createMockEvent)();

    var onChangeSpy = _sinon["default"].spy();

    var mockProps = {
      checked: true,
      name: "someValue",
      id: "test",
      onChange: onChangeSpy
    };
    var custom = (0, _enzyme.shallow)(_react["default"].createElement(_["default"], mockProps));
    it("should call on change prop", function () {
      custom.find('#test').props().onChange(mockEvent);
      var onChangeArgs = [mockEvent, {
        id: mockProps.id,
        name: mockProps.name,
        value: !mockProps.checked
      }];
      expect(onChangeSpy.args[0]).toEqual(onChangeArgs);
    });
  });
  describe("Checkbox rendering", function () {
    it("should match snapshot", function () {
      var wrapper = (0, _enzyme.render)(_react["default"].createElement(_["default"], null));
      expect(wrapper).toMatchSnapshot();
    });
    it("should render the 'as' prop", function () {
      var wrapper = (0, _enzyme.mount)(_react["default"].createElement(_["default"], {
        as: _toolbox.StubComponent
      }));
      expect(wrapper.find(_toolbox.StubComponent).length).toBe(1);
    });
  });
});