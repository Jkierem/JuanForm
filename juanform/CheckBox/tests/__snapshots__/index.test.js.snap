// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Checkbox Checkbox rendering should match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "defaults__CheckBox-sc-157ahxh-1 hSlFKl",
      "type": "checkbox",
    },
    "children": Array [],
    "name": "input",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "type": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "type": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
